python/cugraph/cugraph/link_prediction/jaccard.py

```diff
@@ -108,55 +142,75 @@ def jaccard(input_graph, vertex_pair=None, do_expensive_check=True):
-            expected_nodes = cudf.Series(range(0, max_vertex + 1, 1)).astype(
-                input_df.dtypes[0]
-            )
-            nodes = (
-                cudf.concat([input_df["src"], input_df["dst"]])
-                .unique()
-                .sort_values()
-                .reset_index(drop=True)
-            )
-            if not expected_nodes.equals(nodes):
-                raise ValueError("Unrenumbered vertices are not supported.")
+        warnings.warn(
+            "do_expensive_check is deprecated since it is no longer needed",
+            DeprecationWarning,
+        )
 
     if input_graph.is_directed():
         raise ValueError("Input must be an undirected Graph.")
-    if type(vertex_pair) == cudf.DataFrame:
+
+    if vertex_pair is None:
+        # Call two_hop neighbor of the entire graph
+        vertex_pair = input_graph.get_two_hop_neighbors()
+
+    v_p_num_col = len(vertex_pair.columns)
+
+    if isinstance(vertex_pair, cudf.DataFrame):
         vertex_pair = renumber_vertex_pair(input_graph, vertex_pair)
+        vertex_pair = ensure_valid_dtype(input_graph, vertex_pair)
+        src_col_name = vertex_pair.columns[0]
+        dst_col_name = vertex_pair.columns[1]
+        first = vertex_pair[src_col_name]
+        second = vertex_pair[dst_col_name]
+
     elif vertex_pair is not None:
-        raise ValueError("vertex_pair must be a cudf dataframe")
+        raise ValueError("vertex_pair must be a cudf Dataframe")
 
-    df = jaccard_wrapper.jaccard(input_graph, None, vertex_pair)
+    first, second, jaccard_coeff = pylibcugraph_jaccard_coefficients(
+        resource_handle=ResourceHandle(),
+        graph=input_graph._plc_graph,
+        first=first,
+        second=second,
+        use_weight=use_weight,
+        do_expensive_check=False,
+    )
 
     if input_graph.renumbered:
-        df = input_graph.unrenumber(df, "first")
-        df = input_graph.unrenumber(df, "second")
+        vertex_pair = input_graph.unrenumber(
+            vertex_pair, src_col_name, preserve_order=True
+        )
+        vertex_pair = input_graph.unrenumber(
+            vertex_pair, dst_col_name, preserve_order=True
+        )
+
+    if v_p_num_col == 2:
+        # single column vertex
+        vertex_pair = vertex_pair.rename(
+            columns={src_col_name: "first", dst_col_name: "second"}
+        )
+
+    df = vertex_pair
+    df["jaccard_coeff"] = cudf.Series(jaccard_coeff)
 
     return df
 
 
-def jaccard_coefficient(G, ebunch=None, do_expensive_check=True):
+def jaccard_coefficient(G, ebunch=None, do_expensive_check=False):
     """
     For NetworkX Compatability.  See `jaccard`
 
-    NOTE: This algorithm doesn't currently support datasets with vertices that
-    are not (re)numebred vertices from 0 to V-1 where V is the total number of
-    vertices as this creates isolated vertices.
-
     Parameters
     ----------
-    graph : cugraph.Graph
+    G : cugraph.Graph
```

Comment:
There is a mistake in this docstring. This should be
> G : cugraph.Graph or networkx.Graph

Reply:
Regarding `ebunch`, NetworkX takes an `iterable of node pairs` which is a 2 tuple therefore, if the user passes such, we need to convert it to a cudf.DataFrame
> ebunch : cudf.DataFrame or iterable of node pairs, optional (default=None)

And the description of this input parameter needs to be updated